src/colorUtils.ts

```diff
@@ -22,18 +22,17 @@ import { Colors, ColorScale, DiffColors, isDiffColors, RGBA } from './types';
 
 export const DEFAULT_DIMMED_OPACITY = 0.05;
 
-const opacityFloatToIntegerScale = d3Scale
-  .scaleLinear()
-  .domain([0, 1])
-  .range([0, 255]);
-
 export function opacityFloatToInteger(opacity: number): number {
-  return Math.round(opacityFloatToIntegerScale(opacity));
+  return Math.round(opacity * 255);
 }
 
 export function colorAsArray(color: string): RGBA {
-  const rgbColor = d3Color.rgb(color);
-  return [rgbColor.r, rgbColor.g, rgbColor.b, opacityFloatToInteger(rgbColor.opacity)];
+  const col = d3Color.color(color);
+  if (!col) {
+    throw new Error('invalid color: ' + color);
```

Comment:
I don't think we should crash the app if this happens. Rather return black and `console.warn`.